tests/Kerbero.Integration.Tests/NukiSmartLocksListIntegrationTests.cs

```diff
@@ -0,0 +1,156 @@
+{
+	private static readonly IConfigurationRoot Config = new ConfigurationBuilder()
+		.AddJsonFile("appsettings.Test.json")
+		.AddEnvironmentVariables()
+		.Build();
+
+	private readonly HttpTest _httpTest;
+	private readonly HttpClient _client;
+
+	public NukiSmartLocksListIntegrationTests()
+	{
+		WebApplicationFactory<Program> application = new KerberoWebApplicationFactory<Program>();
+		application.Server.PreserveExecutionContext = true; // fixture for Flurl
+		_httpTest = new HttpTest();
+		_client = application.CreateClient();
+	}
+	
+	public void Dispose()
+	{
+		_httpTest.Dispose();
+		GC.SuppressFinalize(this);
+	}
+
+	[Fact]
+	public async Task GetSmartLocksListByKerberoAccount_Success()
+	{
+		_httpTest.RespondWithJson(
+			new[]
+			{
+				Utilities.NukiJsonSmartlockResponse
+			});
+		
+		var response = await _client.GetAsync("api/nuki/smartlock?accountId=1");
+
+		response.EnsureSuccessStatusCode();
+	}
+	
+	[Fact]
+	public async Task GetSmartLocksListByKerberoAccount_RefreshToken_AndCannotParseTheResponse_Text()
+	{
+		_httpTest.RespondWithJson(new
+		{
+			access_token ="ACCESS_TOKEN", 
+			token_type = "bearer", 
+			expires_in = 2592000, 
+			refresh_token = "REFRESH_TOKEN"
+		}); // from nuki documentation
+		
+		var response = await _client.GetAsync("api/nuki/smartlock?accountId=1");
+
+		response.IsSuccessStatusCode.Should().BeFalse();
+	}
+	
+	[Fact]
+	public async Task GetSmartLocksListByKerberoAccount_Unauthorized_Test()
+	{
+		_httpTest.RespondWith(status: 401, body: System.Text.Json.JsonSerializer.Serialize(new
+		{
+			detailMessage = "Your access token is not authorized",
+			stackTrace = Array.Empty<object>(),
+			suppressedExceptions = Array.Empty<object>()
+		})); 
+		
+		var response = await _client.GetAsync("api/nuki/smartlock?accountId=1");
+
+		response.IsSuccessStatusCode.Should().BeFalse();
+		var resJson = await response.Content.ReadFromJsonAsync<JsonObject>();
+		resJson.Should().NotBeNull();
+		resJson?["message"].Should().NotBeNull();
+	}
+	
+	[Fact]
+	public async Task GetSmartLocksListByKerberoAccount_NoAccount_Test()
+	{
+	
+		var response = await _client.GetAsync("api/nuki/smartlock?accountId=0");
+
+		response.IsSuccessStatusCode.Should().BeFalse();
+		response.StatusCode.Should().Be(HttpStatusCode.Unauthorized);
+		var resJson = await response.Content.ReadFromJsonAsync<JsonObject>();
+		resJson.Should().NotBeNull();
+		resJson?["message"].Should().NotBeNull();
+	}
+	
+	[Fact]
+	public async Task GetSmartLocksListByKerberoAccount_TimeoutNuki_Test()
+	{
+		_httpTest.RespondWith(status: 408); 
+		
+		var response = await _client.GetAsync("api/nuki/smartlock?accountId=1");
+
+		response.IsSuccessStatusCode.Should().BeFalse();
+		response.StatusCode.Should().Be(HttpStatusCode.BadGateway);
+		var resJson = await response.Content.ReadFromJsonAsync<JsonObject>();
+		resJson.Should().NotBeNull();
+		resJson?["message"].Should().NotBeNull();
+	}
+	
+	private class KerberoWebApplicationFactory<TStartup>
```

Comment:
```suggestion
	private class KerberoWebApplicationFactory<TStartup>
```
perchè privata, sinceramente credo sia riutilizzabile in molti test